internal/resources/aws/ebs_snapshot.go

```diff
@@ -0,0 +1,140 @@
+		getSnapshotBlockRequests = decimalPtr(decimal.NewFromInt(*r.MonthlyGetBlockRequests))
+	}
+
+	var putSnapshotBlockRequests *decimal.Decimal
+	if r.MonthlyPutBlockRequests != nil {
+		putSnapshotBlockRequests = decimalPtr(decimal.NewFromInt(*r.MonthlyPutBlockRequests))
+	}
+
+	var fastSnapshotRestoreHours *decimal.Decimal
+	if r.MonthlyPutBlockRequests != nil {
+		fastSnapshotRestoreHours = decimalPtr(decimal.NewFromInt(*r.FastSnapshotRestoreHours))
+	}
+
+	costComponents := []*schema.CostComponent{
+		ebsSnapshotCostComponent(region, gbVal),
+		{
+			Name:            "Fast snapshot restore",
+			Unit:            "DSU-hours",
+			UnitMultiplier:  decimal.NewFromInt(1),
+			MonthlyQuantity: fastSnapshotRestoreHours,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonEC2"),
+				ProductFamily: strPtr("Fast Snapshot Restore"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/EBS:FastSnapshotRestore$/")},
+				},
+			},
+		},
+		{
+			Name:            "ListChangedBlocks & ListSnapshotBlocks API requests",
+			Unit:            "1k requests",
+			UnitMultiplier:  decimal.NewFromInt(1000),
+			MonthlyQuantity: listBlockRequests,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonEC2"),
+				ProductFamily: strPtr("EBS direct API Requests"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/EBS:directAPI.snapshot.List$/")},
+				},
+			},
+		},
+		{
+			Name:            "GetSnapshotBlock API requests",
+			Unit:            "1k SnapshotAPIUnits",
+			UnitMultiplier:  decimal.NewFromInt(1000),
+			MonthlyQuantity: getSnapshotBlockRequests,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonEC2"),
+				ProductFamily: strPtr("EBS direct API Requests"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/EBS:directAPI.snapshot.Get$/")},
+				},
+			},
+		},
+		{
+			Name:            "PutSnapshotBlock API requests",
+			Unit:            "1k SnapshotAPIUnits",
+			UnitMultiplier:  decimal.NewFromInt(1000),
+			MonthlyQuantity: putSnapshotBlockRequests,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonEC2"),
+				ProductFamily: strPtr("EBS direct API Requests"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/EBS:directAPI.snapshot.Put$/")},
+				},
+			},
+		}}
+
+	return &schema.Resource{
+		Name:           r.Address,
+		CostComponents: costComponents, UsageSchema: EbsSnapshotUsageSchema,
+	}
+}
+
+func ebsSnapshotCostComponent(region string, gbVal decimal.Decimal) *schema.CostComponent {
```

Comment:
Can we make this a method on the struct? This way we shouldn't need to pass `region` as an argument as well.

Reply:
Agreed, since no other resource use this method. Fixed.